pkg/processor/reportprocessor.go

```diff
@@ -191,68 +191,105 @@ func getGovernanceResults(dynamicClient dynamic.Interface, clusterInfo types.Man
 	}
 
 	var clusterViolations []*v1alpha2.PolicyReportResult
+
+	// Iterate over policies in the cluster
 	for _, plc := range policyList.Items {
-		plcName := plc.Object["metadata"].(map[string]interface{})["name"].(string)
-		func() {
-			defer func() {
-				if err := recover(); err != nil {
-					glog.V(2).Infof(
-						"Error processing policy %s - expected missing data to be present",
-						plcName,
-					)
+		// Parse relevant fields
+		plcName := plc.GetName()
+		md, _, err := unstructured.NestedMap(plc.Object, "metadata")
+		glog.Warningf("error parsing metadata as a map for policy %s: %s", plcName, err)
+		status, _, err := unstructured.NestedMap(plc.Object, "status")
+		glog.Warningf("error parsing status as a map for policy %s: %s", plcName, err)
+		details, _, err := unstructured.NestedSlice(status, "details")
+		glog.V(1).Infof("error parsing status details as a map for policy %s: %s", plcName, err)
+
+		// Find non-compliant policies
+		if status["compliant"].(string) == "NonCompliant" {
+			// Generate a cluster policy report result for each non-compliant policy within the root policy
+			for idx, detail := range details {
+				var detailMap map[string]interface{}
+				var ok bool
+				if detailMap, ok = detail.(map[string]interface{}); !ok {
+					glog.Warningf("failed to parse history detail as a map for policy %s, template %d", plcName, idx)
+
+					continue
 				}
-			}()
-
-			md := plc.Object["metadata"].(map[string]interface{})
-			status := plc.Object["status"].(map[string]interface{})
-			details := status["details"].([]interface{})
-			if status["compliant"].(string) == "NonCompliant" {
-				for _, detail := range details {
-					if detail.(map[string]interface{})["compliant"] == "NonCompliant" {
-						templateMeta := detail.(map[string]interface{})["templateMeta"].(map[string]interface{})
-						historyItem := detail.(map[string]interface{})["history"].([]interface{})[0].(map[string]interface{})
-						annotations := md["annotations"].(map[string]interface{})
-						category := ""
-						if _, ok := annotations["policy.open-cluster-management.io/categories"]; ok {
-							category = annotations["policy.open-cluster-management.io/categories"].(string)
-						}
-						clusterViolations = append(clusterViolations, &v1alpha2.PolicyReportResult{
-							Policy:      md["name"].(string),
-							Description: historyItem["message"].(string),
-							Scored:      false,
-							Category:    category,
-							Source:      "grc",
-							Timestamp:   metav1.Timestamp{Seconds: time.Now().Unix(), Nanos: int32(time.Now().UnixNano())},
-							Result:      "fail",
-							Properties: map[string]string{
-								"created_at": md["creationTimestamp"].(string),
-								"total_risk": convertSevFromGovernance(getSevFromTemplate(plc, templateMeta["name"].(string))),
-							},
-						})
-					}
+
+				// Find non-compliant wrapped policy templates
+				if detailMap["compliant"] == "NonCompliant" {
```

Comment:
Optional: I find this syntax cleaner to avoid all that indentation. Same with the `if status["compliant"].(string) == "NonCompliant" {` line.

```go
if detailMap["compliant"] != "NonCompliant" {
continue
}
```